endpoints/calendar/index.js

```diff
@@ -1,11 +1,57 @@
 import app from '../../server'
 import fridagar from 'fridagar'
 import makeDebug from 'debug'
-import moment from 'moment'
-import { range } from 'lodash'
+import { range, isString } from 'lodash'
 
 const debug = makeDebug('endpoint:calendar')
 
+const canBeInt = intLike => {
+  const num = Number.parseInt(intLike, 10)
+  return !Number.isNaN(num)
+}
+
+export const normalizeParams = (year, month, day) => {
+  // If string parsing failed, reject the promise
+  if (isString(year) && !canBeInt(year)) return { error: 'Year must be a number' }
+  if (isString(month) && !canBeInt(month)) return { error: 'Month must be a number' }
+  if (isString(day) && !canBeInt(day)) return { error: 'Day must be a number' }
+
+  return { year, month, day }
+}
+
+const lookupHolidays = (yearStr, monthStr, dayStr) => new Promise((resolve, reject) => {
+  const { year, month, day, error } = normalizeParams(yearStr, monthStr, dayStr)
+
+  // Reject promise with relevant error when in error states
+  if (error) reject({ error })
+  if (!year) reject({ error: 'No year was provided' })
+  if (!year && !month && !day) reject({ error: 'No parameters were provided' })
+
+  if (year && !month && !day) {
+    // Year
+    const holidays = range(1, 13).reduce((sum, current) => {
+      debug(`Getting year: ${year}, month: ${current}`)
+      return sum.concat(fridagar.getHolidays(year, current))
+    }, [])
+    resolve(holidays)
+  } else if (year && month && !day) {
+    // Year, Month
+    resolve(fridagar.getHolidays(year, month))
+  } else if (year && month && day) {
+    // Year, Month, Day
+    const holiday = fridagar.getHolidays(year, month).find((current) => {
+      return current.date.toISOString().startsWith(`${year}-${month}-${day}`)
+    })
+    const results = holiday || {
+      date: `${year}-${month}-${day}T00:00:00.000Z`,
```

Comment:
This is btw a totally different date format that the other cases 😄 
Lets fix this before merge.